package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh

```diff
@@ -511,13 +511,13 @@ mac80211_hostapd_setup_bss() {
 	json_get_vars wds wds_bridge dtim_period max_listen_int start_disabled
 
 	set_default wds 0
-	set_default start_disabled 0
-
 	[ "$wds" -gt 0 ] && {
 		append hostapd_cfg "wds_sta=1" "$N"
 		[ -n "$wds_bridge" ] && append hostapd_cfg "wds_bridge=$wds_bridge" "$N"
 	}
-	[ "$staidx" -gt 0 -o "$start_disabled" -eq 1 ] && append hostapd_cfg "start_disabled=1" "$N"
+
+	[ "$staidx" -gt 0 ] && set_default start_disabled 1 || set_default start_disabled 0
```

Comment:
wonder if shellcheck would warn that A && B || C is not if else . Probably better to use an if statement just in case.

Reply:
Interesting, what's shellcheck?

I know that it's not if-else but set_default would never return false, I tried it.

Reply:
https://www.shellcheck.net/wiki/SC2015

shellcheck is a static analyzer for shell files. These files are not really checked and full of shellcheck warnings.

Reply:
Hmm, is there any work being done about rewriting this in ucode? 😅

Reply:
doubt it. Would be an interesting PR.

Reply:
> Hmm, is there any work being done about rewriting this in ucode? 😅

that would be really funny!